localstack/services/cloudformation/engine/transformers.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from typing import Dict, Type, Union
+
+from localstack.utils import testutil
+from localstack.utils.aws import aws_stack
+from localstack.utils.objects import recurse_object
+
+LOG = logging.getLogger(__name__)
+
+TransformResult = Union[dict, str]
+
+
+class Transformer:
+    """Abstract class for Fn::Transform intrinsic functions"""
+
+    def transform(self, parameters: dict) -> TransformResult:
+        """Apply the transformer to the given parameters and return the modified construct"""
+
+
+class AwsIncludeTransformer(Transformer):
+    """Implements the 'AWS::Include' transform intrinsic function"""
+
+    def transform(self, parameters: dict) -> TransformResult:
+        from localstack.services.cloudformation.engine.template_preparer import parse_template
+
+        location = parameters.get("Location")
+        if location and location.startswith("s3://"):
+            s3_client = aws_stack.connect_to_resource("s3")
+            bucket, _, path = location.removeprefix("s3://").partition("/")
+            content = testutil.download_s3_object(s3_client, bucket, path)
+            content = parse_template(content)
+            return content
+        else:
+            LOG.warning("Unexpected Location parameter for AWS::Include transformer: %s", location)
+        return parameters
+
+
+# maps transformer names to implementing classes
+transformers: Dict[str, Type] = {"AWS::Include": AwsIncludeTransformer}
+
+
+def apply_transform_intrinsic_functions(template: dict, stack=None) -> dict:
```

Comment:
While I understand why we currently need it, I'm not super happy about the stack class being in here again :grimacing: